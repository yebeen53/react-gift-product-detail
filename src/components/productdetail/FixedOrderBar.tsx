import WishButton from './WishButton';
import theme from '@/data/theme';
import { useNavigate } from 'react-router-dom';

interface FixedOrderBarProps {
  productId: string;
}

const FixedOrderBar = ({ productId }: FixedOrderBarProps) => {
  const navigate = useNavigate();

  const handleOrderClick = () => {
    navigate(`/order/${productId}`);
  };

  return (
    <>
      <div style={{ height: theme.spacing.spacing16 }} />

      <div
        style={{
          position: 'fixed',
          bottom: 0,
          left: 0,
          right: 0,
          backgroundColor: theme.colors.semantic.backgroundDefault,
          borderTop: '1px solid #ddd',
          display: 'flex',
          justifyContent: 'center',
          padding: theme.spacing.spacing4,
          zIndex: 1000,
        }}
      >
        <div
          style={{
            display: 'flex',
            alignItems: 'center',
            gap: theme.spacing.spacing4,
            width: '100%',
            maxWidth: '720px',
          }}
        >
          <WishButton productId={productId} />
          <button
            style={{
              backgroundColor: theme.colors.semantic.kakaoYellow,
              color: theme.colors.semantic.textDefault,
              flex: 1,
              border: 'none',
              height: theme.spacing.spacing10,
              borderRadius: theme.spacing.spacing1,
              cursor: 'pointer',
            }}
            onClick={handleOrderClick}
          >
            주문하기
          </button>
        </div>
      </div>
    </>
  );
};

export default FixedOrderBar;
