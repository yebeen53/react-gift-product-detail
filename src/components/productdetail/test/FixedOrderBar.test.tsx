import { render, screen, fireEvent } from '@testing-library/react';
import FixedOrderBar from '../FixedOrderBar';
import { BrowserRouter, useNavigate } from 'react-router-dom';


jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), 
  useNavigate: jest.fn(), 
}));


jest.mock('../WishButton', () => ({
  __esModule: true,
  default: ({ productId }: { productId: string }) => (
    <button data-testid="mock-wish-button">찜하기 {productId}</button>
  ),
}));

describe('FixedOrderBar 컴포넌트', () => {
  const mockNavigate = jest.fn();

  beforeEach(() => {
    (useNavigate as jest.Mock).mockReturnValue(mockNavigate); 
    mockNavigate.mockClear();
  });

  it('주문하기 버튼이 올바르게 렌더링되어야 한다', () => {
    render(<FixedOrderBar productId="test-product-id" />, { wrapper: BrowserRouter }); 
    expect(screen.getByRole('button', { name: '주문하기' })).toBeInTheDocument();
  });

  it('찜하기 버튼(모킹)이 올바르게 렌더링되어야 한다', () => {
    render(<FixedOrderBar productId="test-product-id" />, { wrapper: BrowserRouter });
    expect(screen.getByTestId('mock-wish-button')).toBeInTheDocument();
    expect(screen.getByTestId('mock-wish-button')).toHaveTextContent('찜하기 test-product-id');
  });

  it('주문하기 버튼 클릭 시 navigate 함수가 올바른 경로로 호출되어야 한다', () => {
    const productId = 'product-123';
    render(<FixedOrderBar productId={productId} />, { wrapper: BrowserRouter });

    const orderButton = screen.getByRole('button', { name: '주문하기' });
    fireEvent.click(orderButton);

    expect(mockNavigate).toHaveBeenCalledTimes(1);
    expect(mockNavigate).toHaveBeenCalledWith(`/order/${productId}`);
  });

  it('컴포넌트의 스타일 및 구조가 스냅샷과 일치해야 한다', () => {
    const { asFragment } = render(<FixedOrderBar productId="test-product-id" />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});