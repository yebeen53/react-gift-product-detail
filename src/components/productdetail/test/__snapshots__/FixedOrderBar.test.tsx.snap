// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FixedOrderBar 컴포넌트 컴포넌트의 스타일 및 구조가 스냅샷과 일치해야 한다 1`] = `
<DocumentFragment>
  <div
    style="height: 64px;"
  />
  <div
    style="position: fixed; bottom: 0px; left: 0px; right: 0px; background-color: rgb(255, 255, 255); border-top: 1px solid rgb(221, 221, 221); display: flex; justify-content: center; padding: 16px; z-index: 1000;"
  >
    <div
      style="display: flex; align-items: center; gap: 16px; width: 100%; max-width: 720px;"
    >
      <button
        data-testid="mock-wish-button"
      >
        찜하기 test-product-id
      </button>
      <button
        style="background-color: rgb(254, 229, 0); color: rgb(42, 48, 56); flex: 1 1 0%; height: 40px; border-radius: 4px; cursor: pointer;"
      >
        주문하기
      </button>
    </div>
  </div>
</DocumentFragment>
`;
